components/dialog/dialog.js

```diff
@@ -201,18 +198,26 @@ class Dialog extends PropertyRequiredMixin(LocalizeCoreElement(AsyncContainerMix
 		);
 	}
 
-	updated(changedProperties) {
+	async updated(changedProperties) {
 		super.updated(changedProperties);
-		if (!changedProperties.has('asyncState')) return;
-		if (this.asyncState === asyncStates.complete) {
-			this.resize();
+
+		if (changedProperties.has('asyncState') && this.asyncState === asyncStates.complete) {
+			await this._handleAsyncChildren();
+			setTimeout(() => {
+				this.resize();
+			}, 0);
 		}
 	}
 
 	_abort() {
 		this._close('abort');
 	}
 
+	async _handleAsyncChildren() {
+		const composedChildren = getComposedChildren(this, this.predicate);
+		await Promise.all(composedChildren.map(child => waitForElem(child, this.predicate)));
```

Comment:
Agreed with Dave's comment about the predicate being private until we need to expose it, and in this case because we know we don't want to search inside nested dialogs, I was imagining it would be something like:
```javascript
_waitForContent() {
    return waitForElem(
        this.shadowRoot.querySelector('.d2l-dialog-inner'), 
        elem => elem.tagName === 'D2L-DIALOG'
    );
}
```
We could also check the `opened` property but I don't think we care, since an open nested dialog should not have any bearing on the size of the parent dialog.

We might be able to target just the content, but it's probably safest to start at an element that includes the header and footer as well.

Reply:
My edits above were to the method name. I think it is worthwhile being specific, because this is different from waiting for async content, where the consumer is doing server calls to fetch data. This wait method has no idea about that stuff.